### PR TITLE
add h3/3.6.3

### PR DESCRIPTION
**Pull request by SpaceIm**
Specify library name and version:  **h3/3.6.3**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/h3/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    topics = ("conan", "h3", "hierarchical", "geospatial", "indexing")
+    homepage = "https://github.com/uber/h3"
+    url = "https://github.com/conan-io/conan-center-index"
+    exports_sources = "CMakeLists.txt"
+    generators = "cmake"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "h3_prefix": "ANY"
```
Comment (uilianries) [inline recipes/h3/all/conanfile.py]: do you really need it?

File: recipes/h3/all/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,9 @@
+cmake_minimum_required(VERSION 2.8.11)
```
Comment (uilianries) [inline recipes/h3/all/CMakeLists.txt]:
```suggestion
cmake_minimum_required(VERSION 3.1)
```

File: recipes/h3/all/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,9 @@
...
+project(cmake_wrapper)
+
+include(conanbuildinfo.cmake)
+conan_basic_setup()
+
+set(CMAKE_C_STANDARD 99)
```
Comment (uilianries) [inline recipes/h3/all/CMakeLists.txt]:
```suggestion
```

File: recipes/h3/all/CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,9 @@
...
+
+add_subdirectory("source_subfolder")
```
Comment (uilianries) [inline recipes/h3/all/CMakeLists.txt]:
```suggestion
add_subdirectory("source_subfolder")
set_target_properties(h3 PROPERTIES C_STANDARD 99)
```

File: recipes/h3/all/conanfile.py [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+        cmake = self._configure_cmake()
+        cmake.install()
+        tools.rmdir(os.path.join(self.package_folder, "lib", "cmake"))
+
+    def package_info(self):
+        self.cpp_info.libs = tools.collect_libs(self)
+        self.cpp_info.defines.append("H3_PREFIX={}".format(self.options.h3_prefix))
+        if self.settings.os == "Linux":
+            self.cpp_info.system_libs.append("m")
+        self.cpp_info.names["cmake_find_package"] = "h3"
```
Comment (uilianries) [inline recipes/h3/all/conanfile.py]:
```suggestion
```

By default it uses the package name.

File: recipes/h3/all/conanfile.py [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+        self.cpp_info.names["cmake_find_package_multi"] = "h3"
```
Comment (uilianries) [inline recipes/h3/all/conanfile.py]:
```suggestion
```
By default it uses the package name.

File: recipes/h3/all/conanfile.py (hunk 5 above)
Comment (SpaceIm) [reply recipes/h3/all/conanfile.py]: yes but H3 CMakeLists exports target h3, therefore isn't it better to be explicit?

File: recipes/h3/all/conanfile.py (hunk 1 above)
Comment (SpaceIm) [reply recipes/h3/all/conanfile.py]: Well, it's an option provided by the library. The default value disables prefix. I assume that consumers may expect this option even if it's an hack (but not a conan hack).

File: recipes/h3/all/conanfile.py (hunk 5 above)
Comment (uilianries) [reply recipes/h3/all/conanfile.py]: by default Conan exports the package name as target name, which means, h3. Thus, you don't need to export it once it's already defined by conan behavior.

**Comment by SpaceIm** [comment]
@danimtb Why CI reports failure now?

**Comment by danimtb** [comment]
@SpaceIm sorry for that, we force a rescan of the repo and had to cancel many PRs resulting on red report. Let me rerun the job.

**Pull request merged**